xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -479,11 +481,13 @@ void CSelectionStreams::Update(std::shared_ptr<CDVDInputStream> input, CDVDDemux
       s.width = info.width;
       s.height = info.height;
       s.codec = info.codecName;
+      s.demuxerId = -1;
       s.name = StringUtils::Format("%s %i", g_localizeStrings.Get(38032).c_str(), i);
       Update(s);
     }
   }
-  else if(demuxer)
```

Comment:
this doesn't look right. parameters are overwritten, others like filename are set but shouldn't

Reply:
@FernetMenta : Any hints on the proper approach here?

Reply:
In the long run you should get rid of everything related to DVDSTREAM_TYPE_DVD, STREAM_SOURCE_NAV, and SDVDInfo in VideoPlayer. Most (if not all) of the callback CVideoPlayer::OnDiscNavResult should go as well. This should be done directly in InputStream.
I think a good starting point to to eliminate m_dvd.iSelected*Stream and the IsBetterStream crap. Stream changes should be handled by InputStream and demuxer and signalled to VideoPlayer by DMX_SPECIALID_STREAMCHANGE and CDemuxStream::changes

Reply:
@FernetMenta : Since this is not my turf, I'll leave this to someone with more knowledge about VideoPlayer. I justed wanted to get the media flags working again for DVDs

Reply:
> this doesn't look right. parameters are overwritten, others like filename are set but shouldn't

@FernetMenta : I just had another look at this. But it seems to me nothing is overwritten, there's only extra info added since the mux info is added as STREAM_SOURCE_VIDEOMUX and dvd nav info is added as STREAM_SOURCE_NAV. Or am I missing something?

Reply:
@arnova i.e. aspect_ratio, width, height are overwritten with hints coming from demuxer. You also overwrite source and id, which I don't think does any good.